Initial device registration policy - Preview docs (#12257)
microsoftgraph/microsoft-graph-docs: api-reference/beta/resources/azureadjoinpolicy.md

---
title: "azureAdJoinPolicy resource type"

> description: "Properties and relationships of the azureADJoinPolicy resource"

Reviewer: ```suggestion
description: "Represents the policy scope of an Active Directory tenant that controls device registration using Azure AD Join."
```

author: "spunukol"
localization_priority: Normal
ms.prod: "directory-management"
doc_type: resourcePageType
---

# azureAdJoinPolicy resource type

Namespace: microsoft.graph

[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]

> Represents the policy scope of the Azure Active Directory tenant that controls the ability for users and groups to register device identities to your organization using Azure AD Join. Only the read and update operations are supported on this resource; create and delete are not supported.

Reviewer: ```suggestion
Represents the policy scope of the Azure Active Directory tenant that controls the ability for users and groups to register device identities to your organization using Azure AD Join.
```

> 

Reviewer: Should be a section titled Methods and in that section should be a table listing the Get and Update methods with links to the method pages. See other published resource pages for examples.

Method | Return Type | Description
-- | -- | --

## Properties
|Property|Type|Description|
|:---|:---|:---|
|allowedGroups|String collection|Group IDs in scope of policy. Required when used with the `selected` value in the `appliesTo` property.|
|allowedUsers|String collection|User IDs in scope of policy. Required when used with the `selected` value in the `appliesTo` property.|
|appliesTo|policyScope|Specifies whether to block, allow or fine-grained control of the policy scope. Possible values are: `none`, `all`, `selected`, `unknownFutureValue`. Required.|
|isAdminConfigurable|Boolean|Specifies whether this `policyScope` is configurable by the admin. Read-only.|

> ## Special considerations when using `appliesTo`

Reviewer: This should not be a new section. The information should just be a continuation of the information in the properties section.

---
title: "azureAdJoinPolicy resource type"

> description: "Represents the policy scope of an Active Directory tenant that controls device registration using Azure AD Join."

Reviewer 2: Kindly confirm: is the join type called Azure AD _Joined_ or Azure AD _Join_? [This article you linked to](https://docs.microsoft.com/en-us/azure/active-directory/devices/overview) seems to indicate **Azure AD joined**.

Author: device identity nomenclature is Azure AD Join but he post the actual registration the device is stamped as Azure AD joined (that is on the object itself). This is good as it is in my opinion

author: "spunukol"
localization_priority: Normal
ms.prod: "directory-management"
doc_type: resourcePageType
---
# azureAdJoinPolicy resource type

Namespace: microsoft.graph

[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]

> Represents the policy scope of the Azure Active Directory tenant that controls the ability for users and groups to register device identities to your organization using Azure AD Join.

Reviewer 2: Same comment as before: Azure AD _Joined_ or Azure AD _Join_

Author: same comment as earlier for this one too

> ## Methods
> 
> | Method | Return Type | Description |
> | --- | --- | --- |
> | [Get deviceRegistrationPolicy](../api/deviceregistrationpolicy-get.md) | [azureAdJoinPolicy resource type](azureadjoinpolicy.md) | Read the properties of a **deviceRegistrationPolicy** object. |
> | [Update deviceRegistrationPolicy](../api/deviceregistrationpolicy-update.md) | [azureAdJoinPolicy resource type](azureadjoinpolicy.md) | Update the properties of a **deviceRegistrationPolicy** object. |
> 

Reviewer 2: No methods for complex type. These methods are for the **deviceRegistrationPolicy** entity.
```suggestion
```

## Properties

|Property|Type|Description|
|:---|:---|:---|

> |allowedGroups|String collection|Group IDs in scope of policy. Required when used with the `selected` value in the `appliesTo` property.|
> |allowedUsers|String collection|User IDs in scope of policy. Required when used with the `selected` value in the `appliesTo` property.|
> |appliesTo|policyScope|Specifies whether to block, allow or fine-grained control of the policy scope. Possible values are: `none`, `all`, `selected`, `unknownFutureValue`. Required.|
> |isAdminConfigurable|Boolean|Specifies whether this `policyScope` is configurable by the admin. Read-only.|

Reviewer 2: A few observations:
- appliesTo property returns 0,1 ... string values not `none`, `all`, `selected`.
  - Ideally, customers always expect the API to return the string name, not its int representation. Should changes be made to the schema before the docs are published?
- isAdminConfigurable property is updatable.
- None of the properties is required.

```suggestion
|allowedGroups|String collection|The identifiers of the groups that are in the scope of the policy. Required when the **appliesTo** property is set to `selected`. |
|allowedUsers|String collection|The identifiers of users that are in the scope of the policy. Required when the **appliesTo** property is set to `selected`.|
|appliesTo|policyScope|Specifies whether to block or allow fine-grained control of the policy scope. Possible values are: `none`, `all`, `selected`, `unknownFutureValue`. The default value is `0`. When set to `2`, at least one user or group identifier must be specified in either **allowedUsers** or **allowedGroups**.  Setting this property to `0` or `1` removes all identifiers in both **allowedUsers** and **allowedGroups**.|
|isAdminConfigurable|Boolean|Specifies whether this policy scope is configurable by the admin. The default value is `false`.|
```
